cli/cppcheckexecutor.cpp

```diff
@@ -170,9 +174,14 @@ namespace {
         std::set<std::string> mActiveCheckers;
 
         /**
-         * True if there are critical errors
+         * List of critical errors
          */
         std::string mCriticalErrors;
+
+        /**
+         * CTU information
+         */
+        std::string mCtuInfo;
```

Comment:
Is this safe in terms of memory usage? I have not really understood what this is doing yet (maybe add some explanation to the PR) but appears to accumulate the the data in the memory and this could be megabytes in size or even much, much more.

Reply:
we should check it but my hypothesis is that the ctu info will not be huge.

Reply:
about memory usage, here is one small test case: cppcheck/test/cli/whole-program
 * the files whole1.c and whole2.c are 94 and 64 bytes.
 * the ctu-info that is generated for those are 220 bytes and 227 bytes.

in this test case ~160 bytes source code generates ~450 bytes ctu-info

I don't think that memory usage will be a large issue. If we pretend that scanning a large project with 160MB source code would require 450MB memory for ctu info.

Reply:
Thanks.

But that has to be constructed as a string and is required to be continuous memory.

And we also need to profile that. That sounds like it might slow down things quite a bit. Maybe we might need both modes?